src/Microsoft.Diagnostics.Monitoring.Egress/IEgressEndpoint.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.Egress
+{
+    internal interface IEgressEndpoint<TOptions>
```

Comment:
I don't feel very strongly about this, and maybe this is just me, but when I think of IEgressEndpoint, I keep thinking of the .net Endpoint hierarchy (such as IPEndPoint). How about IDataEgress, or IEgressStrategy, or just IEgress?

Reply:
Renamed to EgressProvider.